src/evict/evict_lru.c

```diff
@@ -1720,10 +1720,22 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 			 * Try a different walk start point next time if a
 			 * walk gave up.
 			 */
-			btree->evict_walk_state =
```

Comment:
I pushed something that keeps the arithmetic with minimal casting.

@keithbostic, is that approach okay with you?